pkg/git/git.go

```diff
@@ -5,10 +5,7 @@ import (
 	"fmt"
 )
 
-// Provider acts as an interface for specific Git hosting providers -- e.g. GitHub, BitBucket, GitLab, etc.
-// It wraps a low-level git implementation (e.g. gogit) and abstracts auth and provider specific configurations
-// while providing a common interface for local git actions.
-type Provider interface {
+type Client interface {
```

Comment:
How many implementations of 'Client' are there?

Reply:
right now, there's a single implementation of Client -- using `go-git`. The only consumer at this point is the Flux system (misnamed `FluxAddonClient`) and the accompanying e2e tests. The thought behind having Git behind a  specific interface was that it would permit us to switch out the implementation underneath is more easily as needed -- while we're using the golang library [go-git](https://github.com/go-git/go-git) there was some discussion initially about wether or not to use use a git executable. If we ever wanted to swap this out we could do it pretty easily

the Provider interface (Client and Provider were one interface until just a few days ago) will potentially have more implementations as we support more git providers (gitlab, etc) as first-class citizens; right now, it's just Github.

Reply:
>  it would permit us to switch out the implementation underneath it more easily 

This is a misconception. Duck typing renders this interface redundant so it can be removed entirely given your summary.